File: .github/workflows/list-resources.py
import yaml

template_path = 'template.yml'

with open(template_path, 'r') as f:
    template = yaml.safe_load(f)

resources = template.get('Resources', {})
resource_types = [resource['Type'].split('::')[-1].strip().split(':')[-1].strip() for resource in resources.values()]

with open('resources.txt', 'w') as f:
    f.write('\n'.join(resource_types))

File: list-resources.js
const fs = require('fs');
const yaml = require('yaml');

const templatePath = 'template.yml';
const templateContents = fs.readFileSync(templatePath, 'utf8');
const template = yaml.parse(templateContents);

const resources = Object.keys(template.Resources || {});

console.log(resources.join('\n'));
console.log("::set-output name=resources::" + JSON.stringify(resources));
process.stdout.write(`::set-output name=resources::${JSON.stringify(resources)}`);
